### PR TITLE
BUILD-2179 Use Vault

### PR DESCRIPTION
**Pull request by Godin**
- [x] depends on https://github.com/SonarSource/re-terraform-aws-vault/pull/272
- [x] `.cirrus.yml`
- [x] `.github/workflows/release.yml`

### CODE REVIEW

**Comment by tomverin** [comment]
In release.yml, jfrog cli needs to be configured with an access token like [here](https://github.com/SonarSource/gh-action_release/blob/master/.github/workflows/maven-central.yaml#L54)

**Comment by tomverin** [comment]
in .cirrus.yml the cli needs to be configured in multiple steps in order to use an access token and not a pre-configured config with an api key

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -12,26 +12,39 @@ jobs:
...
       - uses: actions/setup-node@v1
         with:
-          node-version: 12      
-      - run: mv .github/workflows/.npmrc .npmrc       
-      - name: Publish npm package            
+          node-version: 12
+      - run: mv .github/workflows/.npmrc .npmrc
+      - name: Publish npm package
         env:
           NPM_TOKEN: ${{ secrets.SONARTECH_NPM_TOKEN }}
```
Comment (Godin) [inline .github/workflows/release.yml:35]: `NPM_TOKEN` should be taken from Vault.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -12,26 +12,39 @@ jobs:
...
           REPOX_CLI_CONFIG_BUILD_PROMOTER_LT: ${{ secrets.REPOX_CLI_CONFIG_BUILD_PROMOTER_LT }}
```
Comment (Godin) [inline .github/workflows/release.yml:36]: @tomverin shouldn't `REPOX_CLI_CONFIG_BUILD_PROMOTER_LT` be taken from Vault?

File: .cirrus.yml [hunk 3]
```diff
@@ -1,10 +1,15 @@
...
 env:
-  GITHUB_TOKEN: ENCRYPTED[!f458126aa9ed2ac526f220c5acb51dd9cc255726b34761a56fc78d4294c11089502a882888cef0ca7dd4085e72e611a5!]
-  SONAR_TOKEN: ENCRYPTED[!4e5894b7b32f763db69e70fb56188d3f42539db36374b1f97c425f37a7ecd441c28a9da267c461b9cc8fb53124f14c22!]
-  REPOX_CLI_CONFIG_QA_DEPLOYER: ENCRYPTED[!07d065a3727e6f270943618078b68718ff229941f1133c92e587f8e258c086f50f61a75f99496a8b2d7045da177b3ad1!]
-  REPOX_CLI_CONFIG_BUILD_PROMOTER: ENCRYPTED[!a4c9e5455d42b4395387b7489c47bb4c70e85cae5641b77b3b7e53b68ae57bc7b2e591ca4dd640138e15551f2915b08f!]
+  CIRRUS_VAULT_URL: https://vault.sonar.build:8200
+  CIRRUS_VAULT_AUTH_PATH: jwt-cirrusci
+  CIRRUS_VAULT_ROLE: cirrusci-${CIRRUS_REPO_OWNER}-${CIRRUS_REPO_NAME}
+
+  #GITHUB_TOKEN: ENCRYPTED[!f458126aa9ed2ac526f220c5acb51dd9cc255726b34761a56fc78d4294c11089502a882888cef0ca7dd4085e72e611a5!]
```
Comment (Godin) [inline .cirrus.yml:8]: Should be removed.

File: .github/workflows/release.yml (hunk 1 above)
Comment (tomverin) [reply .github/workflows/release.yml:35]: not used anymore, forgot to remove it from env

File: .github/workflows/release.yml (hunk 2 above)
Comment (tomverin) [reply .github/workflows/release.yml:36]: not used anymore, should be removed from env also

File: .cirrus.yml (hunk 3 above)
Comment (tomverin) [reply .cirrus.yml:8]: indeed

**Pull request merged**